src/transformers/pipelines/base.py

```diff
@@ -760,14 +760,21 @@ def __init__(
         self.feature_extractor = feature_extractor
         self.modelcard = modelcard
         self.framework = framework
-        if is_torch_available() and isinstance(device, torch.device):
-            self.device = device
+        if is_torch_available() and self.framework == "pt":
+            if isinstance(device, torch.device):
+                self.device = device
+            elif type(device) == str:
```

Comment:
```suggestion
            elif isinstance(int, str):
```
Will work for both, I believe, and then you can remove the last else (or raise a ValueError there)